lib/BuildSystem/BuildSystemFrontend.cpp

```diff
@@ -792,9 +792,16 @@ bool BuildSystemFrontend::build(StringRef targetToBuild) {
   if (!buildSystem->build(targetToBuild))
     return false;
 
+  bool wasCancelled = false;
+  auto impl = static_cast<BuildSystemFrontendDelegateImpl*>(delegate.impl);
+  if (impl->getStatus() == BuildSystemFrontendDelegateImpl::Status::Cancelled) {
+    wasCancelled = true;
+  }
+
   // The build was successful if there were no failed commands or unspecified
   // errors.
   //
   // It is the job of the client to report a summary, if desired.
-  return delegate.getNumFailedCommands() == 0 && delegate.getNumErrors() == 0;
+  return !wasCancelled && delegate.getNumFailedCommands() == 0
```

Comment:
This was the cause of downstream test failures when #291 was first landed.   In that scenario, the client triggers cancellation immediately and thus no tasks actually start (thus none fail).  Because we previously returned success (i.e. true), the client then proceeded to attempt to validate that all of the commands it expected to run and failed because none of them actually were run.

Reply:
(and in writing this, it reminds me to put this into a comment in the code)

Reply:
👍